MC_data_new_TH3red.py

```diff
@@ -0,0 +1,662 @@
+                param_tree.Branch('EDepHit_og',EDepHit_og,"EDepHit_og[nhits_og]/F")            
+                
+                final_imgs=list();
+                
+                if opt.events==-1:
+                    totev=tree.GetEntries()
+                else:
+                    
+                    if opt.events<=tree.GetEntries():
+                        totev=opt.events
+                    else:
+                        totev=tree.GetEntries()
+                    
+
+    
+                for entry in range(0, totev): #RUNNING ON ENTRIES
+                    tree.GetEntry(entry)
+                    print(entry)
+                    eventnumber[0] = tree.eventnumber
+                    #FIXME
+                    particle_type[0] = 0
+                    energy_ini[0] = tree.ekin_particle[0]*1000
+                    phi_ini[0] = -999.
+                    theta_ini[0] = -999.
+                    phi_ini[0] = np.arctan2( (tree.y_hits[1]-tree.y_hits[0]),(tree.z_hits[1]-tree.z_hits[0]) )
+                    theta_ini[0] = np.arccos( (tree.x_hits[1]-tree.x_hits[0]) / np.sqrt( np.power((tree.x_hits[1]-tree.x_hits[0]),2) + np.power((tree.y_hits[1]-tree.y_hits[0]),2) + np.power((tree.z_hits[1]-tree.z_hits[0]),2)) )
+                    outtree.Fill()
+                    theta[0]=theta_ini[0]
+                    phi[0]=phi_ini[0]
+                    energy[0]=energy_ini[0]
+                    track_length_3D[0]=0.0
+                    proj_track_2D[0]=0.0
+                    nhits_og[0]=tree.numhits
+                    final_image=rt.TH2I('pic_run'+str(run_count)+'_ev'+str(entry), '', opt.x_pix, 0, opt.x_pix-1, opt.y_pix, 0, opt.y_pix-1) #smeared track with background
+
+                    x_hits_tr = tree.x_hits
+
+                    #print("Original Info on X")
+                    #print(x_hits_tr)
+
+                    if opt.randZ:
+                        opt.z_gem=0
+                        rand = 50 + random.random()*400
+                        X0= min(tree.x_hits)
+                        for ihit in range(0,tree.numhits):
+                            x_hits_tr[ihit]-=X0
+                            x_hits_tr[ihit]+=rand
+
+                    #print("Random Diffused")
+                    #print(x_hits_tr)
+                    #print(opt.z_gem)
+                    histname = "histo_cloud_pic_"+str(run_count)+"_ev"+str(int(entry))
+                    
+                                        
+                    yMinCl=1152+(min(tree.y_hits))*opt.y_pix/opt.y_dim-20
+                    yMaxCl=1152+(max(tree.y_hits))*opt.y_pix/opt.y_dim+20
+
+                    zMinCl=1152+(min(tree.z_hits))*opt.x_pix/opt.x_dim-20
+                    zMaxCl=1152+(max(tree.z_hits))*opt.x_pix/opt.x_dim+20
+
+                    xMinCl=((min(x_hits_tr))-10)/opt.z_vox_dim
+                    xMaxCl=((max(x_hits_tr))+10)/opt.z_vox_dim
+
+                    zbins = int((xMaxCl-xMinCl)/opt.z_vox_dim)
```

Comment:
I have compared the output of this code (Samuele's) with the current master branch:

[https://github.com/CYGNUS-RD/digitization/blob/master/MC_data_new.py](url),

and also with this old version:

[https://github.com/CYGNUS-RD/digitization/blob/49d0ee39e68a3d1cb5a2a03ed5d8fea05e3ec060/MC_data_new.py](url)
(this version is more similar to Samuel's since they both use the "old and slow" method for saturation).

The result is that Samuele's code gives images that contain around two times the number of photons in the images produced with the other two versions.
   
Note: In all the 3 simulations I have used the same parameters in the config file, I turned off the random Z for Sameule's code, and I used the same input rootfile, generated by Samuele in Geant4. 

I realized the problem was in the saturation effect, since the difference disappeared when either turning the effect off or setting `beta` to zero. And eventually I noticed that the master code (and the "old and slow" one) were able to reproduce Samuele's images by setting `z_vox_dim` parameter equals to the square of the `z_vox_dim` in Sameule's config file. For instance:

`z_vox_dim` for master code = 0.01 
`z_vox_dim` for Samuele's = 0.1
     
So, I may be wrong, but I think on line 317, `zbins` should be rather defined this way:

`zbins = int(xMaxCl-xMinCl)`

That is, without dividing by `opt.z_vox_dim`, because you already divided by it when defining `xMinCl` and `xMaxCl`

@SamueleTorelli what do you think?